datafusion/optimizer/src/filter_push_down.rs

```diff
@@ -797,7 +817,7 @@ impl OptimizerRule for FilterPushDown {
         plan: &LogicalPlan,
         _: &mut OptimizerConfig,
     ) -> Result<LogicalPlan> {
-        optimize(plan, State::default())
+        optimize(plan, State::default().with_cnf_rewrite())
```

Comment:
@alamb   test fail cause of after https://github.com/apache/arrow-datafusion/pull/3880, it runs rules multiple times.
When after running first time it will produce `filter` front (near `scan`) the `join`, but we don't need write the filter-expr after pushDown (expect page index, i will manually add it), so i add this option.🤔